libraries/botbuilder-core/tests/browserStorage.test.js

```diff
@@ -1,127 +1,112 @@
-        return storage.write({ keyCreate: { count: 1 } })
-            .then(() => storage.read(['keyCreate']))
-            .then((result) => {
-                assert(result != null, 'result should be object');
-                assert(result.keyCreate != null, 'keyCreate should be defined');
-                assert(result.keyCreate.count == 1, 'object should have count of 1');
-                assert(result.keyCreate.eTag, 'ETag should be defined');
-            });
+    it('key creation', async function () {
+        await storage.write({ keyCreate: { count: 1 } });
+        const result = await storage.read(['keyCreate']);
+        assert(result != null, 'result should be object');
+        assert(result.keyCreate != null, 'keyCreate should be defined');
+        assert(result.keyCreate.count === 1, 'object should have count of 1');
+        assert(result.keyCreate.eTag, 'ETag should be defined');
     });
 
-    it('key update', function () {
-        return storage.write({ keyUpdate: { count: 1 } })
-            .then(() => storage.read(['keyUpdate']))
-            .then((result) => {
-                result.keyUpdate.count = 2;
-                return storage.write(result)
-                    .then(() => storage.read(['keyUpdate']))
-                    .then((updated) => {
-                        assert(updated.keyUpdate.count == 2, 'object should be updated');
-                        assert(updated.keyUpdate.eTag != result.keyUpdate.eTag, 'Etag should be updated on write');
-                    });
-            })
+    it('key update', async function () {
+        await storage.write({ keyUpdate: { count: 1 } });
+
+        const result = await storage.read(['keyUpdate']);
+        result.keyUpdate.count = 2;
+        await storage.write(result);
+
+        const updated = await storage.read(['keyUpdate']);
+        assert(updated.keyUpdate.count === 2, 'object should be updated');
+        assert(updated.keyUpdate.eTag !== result.keyUpdate.eTag, 'Etag should be updated on write');
     });
 
-    it('invalid eTag', function () {
-        return storage.write({ keyUpdate2: { count: 1 } })
-            .then(() => storage.read(['keyUpdate2']))
-            .then((result) => {
-                result.keyUpdate2.count = 2;
-                return storage.write(result).then(() => {
-                    result.keyUpdate2.count = 3;
-                    return storage.write(result)
-                        .then(() => assert(false, 'should throw an exception on second write with same etag'))
-                        .catch((reason) => { });
-                });
-            });
+    it('invalid eTag', async function () {
+        await storage.write({ keyUpdate2: { count: 1 } });
+
+        const result = await storage.read(['keyUpdate2']);
+        result.keyUpdate2.count = 2;
+        result.keyUpdate2.eTag = -1;
+
+        await assert.rejects(storage.write(result), Error('Storage: error writing "keyUpdate2" due to eTag conflict.'));
```

Comment:
Note: This is another test that would always pass when it shouldn't. As you can see, I changed `result.keyUpdate2.eTag` intentionally to something that was different than it's previous value. [eTags of the same value are NOT invalid](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-core/src/memoryStorage.ts#L84). However, an new eTag < the old eTag IS invalid. [This is handled the same way in .NET.](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/MemoryStorage.cs#L155)